File: Cpp_src/easy/power_of_two_231.cc
#include <iostream>
using namespace std;

// Use bit manipulation
// Time Complexity: O(1)
// Space Complexity:O(1)
// Runtime: 4 ms, faster than 56.82%
// Memory Usage: 8.1 MB, less than 68.00%
class Solution
{
public:
    bool isPowerOfTwo(int n)
    {
        return ((n > 0) && ((n & (n - 1)) == 0));
    }
};

File: Golang_src/easy/powerOfTwo_231.go
package easy

// Use Bit Manipulation:
// Time Complexity: O(1)
// Space Complexity:O(1)
// Runtime: 4 ms, faster than 42.86%
// Memory Usage: 2.2 MB, less than 100.00%
func isPowerOfTwo(n int) bool {
	return n > 0 && n&(n-1) == 0
}
